hydropandas/observation.py

```diff
@@ -777,6 +777,46 @@ def from_pastastore(cls, pstore, libname, name, metadata_mapping=None):
 
         return cls(data, meta=metadata, **kwargs)
 
+    @classmethod
+    def from_solinst(
+            cls,
+            path,
+            transform_coords=True,
+            screen_bottom=None, screen_top=None, ground_level=None,
+            tube_nr=None, tube_top=None):
+        """Read data from Solinst xle file.
+
+        Parameters
+        ----------
+        path : str
+            path to file (file can zip or xle)
+
+        """
+        from .io import solinst
+
+        df, meta = solinst.read_solinst_file(
+            path,
+            transform_coords=transform_coords
+        )
+
+        return cls(
+            df,
+            name=meta.pop("name"),
```

Comment:
add full metadata dict first, then pop the items?

Reply:
I copied this working order from an other read package.

Reply:
Should probably also be changed there then, I think :). But i modified the order, so it should be fine now.

Reply:
I thought you would want the metadata to be available both in the Obs object and as columns in the ObsCollection?

Reply:
But you say it doesn't matter? Does the dictionary get emptied regardless of the order you provide the keyword arguments?

Reply:
Just to be sure we have the same idea of what is happening: When we use `x = meta.pop('x')` we remove `x` from the `meta` dict and add `x` as an attribute to the obs object. Then we use `meta=meta` at the end to add the meta dict as the attribute `meta` to the Obs object. When we do it in this order the `meta` attribute of an Obs object does not have any keys that are also attributes of that Obs object. 

I like this approach because I don't like to store the same information in multiple places. If we would change the order we might end up with an Obs object that has a `meta` attribute which is a dictionary with a key `x` and an attribute `x` which have the same value.

But then I am not sure what happens if you use a dictionary as the first function argument. I think a reference to that dictionary is passed to the function and not a copy so if you modify the dictionary in other function arguments (by popping items) I think it will have an influence the first argument. This should be easy to test.

I hope this makes it more clear. For now I think we can leave it as is and I can pick this up later

Reply:
Fair enough, I somehow thought it would be nice to have both metadata available as a single dictionary containing everything, and certain attributes for easy quick access. But I get that storing the same data in multiple locations is a bit redundant and maybe more confusing. So all good to keep things as they are, and maybe add a comment why the data is being popped :). For future me, or future someone else.